google-cloud-storage/src/main/java/com/google/cloud/storage/NotificationInfo.java

```diff
@@ -0,0 +1,362 @@
+            @Override
+            public NotificationInfo apply(com.google.api.services.storage.model.Notification pb) {
+              return NotificationInfo.fromPb(pb);
+            }
+          };
+  static final Function<NotificationInfo, com.google.api.services.storage.model.Notification>
+      TO_PB_FUNCTION =
+          new Function<NotificationInfo, com.google.api.services.storage.model.Notification>() {
+            @Override
+            public com.google.api.services.storage.model.Notification apply(
+                NotificationInfo NotificationInfo) {
+              return NotificationInfo.toPb();
+            }
+          };
+
+  public enum PayloadFormat {
+    JSON_API_V1,
+    NONE
+  }
+
+  public enum EventType {
+    OBJECT_FINALIZE,
+    OBJECT_METADATA_UPDATE,
+    OBJECT_DELETE,
+    OBJECT_ARCHIVE
+  }
+
+  private final String generatedId;
+  private final String topic;
+  private final List<EventType> eventTypes;
+  private final Map<String, String> customAttributes;
+  private final PayloadFormat payloadFormat;
+  private final String objectNamePrefix;
+  private final String etag;
+  private final String selfLink;
+
+  /** Builder for {@code NotificationInfo}. */
+  public abstract static class Builder {
+    Builder() {}
+
+    abstract Builder setGeneratedId(String generatedId);
+
+    public abstract Builder setSelfLink(String selfLink);
+
+    public abstract Builder setTopic(String topic);
+
+    public abstract Builder setPayloadFormat(PayloadFormat payloadFormat);
+
+    public abstract Builder setObjectNamePrefix(String objectNamePrefix);
+
+    public abstract Builder setEventTypes(EventType... eventTypes);
+
+    public abstract Builder setEtag(String etag);
+
+    public abstract Builder setCustomAttributes(Map<String, String> customAttributes);
+
+    /** Creates a {@code NotificationInfo} object. */
+    public abstract NotificationInfo build();
+  }
+
+  /** Builder for {@code NotificationInfo}. */
+  public static class BuilderImpl extends Builder {
+
+    private String generatedId;
+    private String topic;
+    private List<EventType> eventTypes;
+    private Map<String, String> customAttributes;
+    private PayloadFormat payloadFormat;
+    private String objectNamePrefix;
+    private String etag;
+    private String selfLink;
+
+    BuilderImpl(String topic) {
+      this.topic = topic;
+    }
+
+    BuilderImpl(NotificationInfo notificationInfo) {
+      generatedId = notificationInfo.generatedId;
+      etag = notificationInfo.etag;
+      selfLink = notificationInfo.selfLink;
+      topic = notificationInfo.topic;
+      eventTypes = notificationInfo.eventTypes;
+      customAttributes = notificationInfo.customAttributes;
+      payloadFormat = notificationInfo.payloadFormat;
+      objectNamePrefix = notificationInfo.objectNamePrefix;
+    }
+
+    @Override
+    Builder setGeneratedId(String generatedId) {
+      this.generatedId = generatedId;
```

Comment:
use `notificationId` instead.
Is this meant to be a String or int?

Reply:
its String